applications/feedback_sentiment_analyzer/SPECIFICATION.md

```diff
@@ -0,0 +1,383 @@
+{
+  "feedback": [
+    {
+      "sentiment": "POSITIVE",
+      "text": "I love this hotel!",
+      "audioUrl": "PXL_20230710_182358532.jpg.mp3",
+      "imageUrl": "PXL_20230710_182358532.jpg"
+    }
+  ]
+}
+</pre>
+</td>  
+</tr>
+</table>
+</details>
+
+<details>
+<summary><strong>GET /api/env</strong></summary>
+<br/>
+
+> Get the environment variables required to connect to a Cognito hosted UI. The frontend calls this automatically to facilitate sign in.
+
+<table>
+<tr>
+<th>parameters</th>
+<th>request body</th>
+<th>response body</th>
+</tr>
+<tr>
+<td>Empty</td>
+<td>Empty</td>
+<td>
+<pre>
+{
+"COGNITO_SIGN_IN_URL": "https://...",
+"COGNITO_SIGN_OUT_URL": "https://..."
+}
+</pre>
+</td>
+</tr>
+</table>
+</details>
+
+## Step Function configuration
+
+When an image is created or updated in S3 media bucket, a Step Function state machine is triggered.
+
+This multi-state workflow is listed below in sequence:
+
+1. Start
+2. [ExtractText](#ExtractText) - extracts text from an image
+3. [AnalyzeSentiment](#AnalyzeSentiment) - detects text sentiment
+4. `ContinueIfPositive` (skip to 7 if sentiment `NEGATIVE`)
+5. [TranslateText](#TranslateText) - translates text to English
+6. [SynthesizeAudio](#SynthesizeAudio) - synthesizes human-like audio from text
+7. `DynamoDB:PutItem` (see [table config](#managing-items-in-dynamodb))
+8. Stop
+
+The below diagram depicts this sequence.
+![state-machine.png](state-machine.png)
+
+## State machine Lambda functions
+
+See below for the required inputs and outputs of each Lambda function.
+
+### ExtractText
+Uses Amazon Textract's [DetectDocumentText](https://docs.aws.amazon.com/textract/latest/dg/API_DetectDocumentText.html)
+method to extract text from an image and return a unified text representation.
+
+#### **Input**
+
+Uses the data available on the [S3 event object](https://docs.aws.amazon.com/AmazonS3/latest/userguide/EventBridge.html).
+
+For example:
+
+```json
+{
+  "bucket": "my-s3-bucket",
+  "region": "us-east-1",
+  "object": "foo/bar.png"
+}
+```
+
+#### **Output**
+
+Returns a string representing the text extracted.
+
+For example:
+
+```
+CET HÔTEL ÉTAIT SUPER
+```
+---
+### AnalyzeSentiment
+Uses Amazon Comprehend's [DetectSentiment](https://docs.aws.amazon.com/comprehend/latest/APIReference/API_DetectSentiment.html)
+method to detect sentiment (`POSITIVE`, `NEUTRAL`, `MIXED`, or `NEGATIVE`).
+
+#### **Input**
+
+Uses the data available on the [Lambda event object](https://docs.aws.amazon.com/lambda/latest/dg/gettingstarted-concepts.html#gettingstarted-concepts-event).
+
+For example:
+
+```json
+{
+  "source_text": "CET HÔTEL ÉTAIT SUPER",
+  "region": "us-east-1"
+}
+```
+
+#### **Output**
+
+Returns the guessed sentiment and language code.
+
+For example:
+
+```json
+{
+  "sentiment": "POSITIVE",
+  "language_code": "fr-FR"
+}
+```
+---
+### TranslateText
+Uses Amazon Translate's [TranslateText](https://docs.aws.amazon.com/translate/latest/APIReference/API_TranslateText.html)
```

Comment:
```suggestion
Uses the Amazon Translate [TranslateText](https://docs.aws.amazon.com/translate/latest/APIReference/API_TranslateText.html)
```

Reply:
Same here as above. Are we missing a word a the end?